server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -381,6 +382,114 @@ func (s *BuildBuddyServer) JoinGroup(ctx context.Context, req *grpb.JoinGroupReq
 	return &grpb.JoinGroupResponse{}, nil
 }
 
+func (s *BuildBuddyServer) GetApiKeys(ctx context.Context, req *akpb.GetApiKeysRequest) (*akpb.GetApiKeysResponse, error) {
+	userDB := s.env.GetUserDB()
+	if userDB == nil {
+		return nil, status.UnimplementedError("Not Implemented")
+	}
+	groupID := req.GetGroupId()
+	if err := s.authorizeGroupAccess(ctx, groupID); err != nil {
+		return nil, err
+	}
+	tableKeys, err := userDB.GetAPIKeys(ctx, groupID)
+	if err != nil {
+		return nil, err
+	}
+	protoKeys := make([]*akpb.ApiKey, len(tableKeys))
```

Comment:
nice -- did you read about the difference between length and capacity?

In this code what you have is just perfect, but if you occasionally expected an error in your for loop that would mean skipping an element, you could instead do:

keys := make([].., 0, len(tableKeys))
for {
  keys = append(keys, blah)
}

and still get most of the benefit of knowing the length ahead of time (aka reduced allocations).

Reply:
Ah cool, I knew there was a difference, somehow missed that you could specify the capacity explicitly in `make` -- done!